torch/nested/_internal/ops.py

```diff
@@ -179,6 +191,28 @@ def raggedness_matches(nt, size):
     )
 
 
+def assert_raggedness_matches(nt, size, msg):
+    from torch.fx.experimental.symbolic_shapes import is_nested_int
+    from torch.nested._internal.nested_int import _nested_assert_metadata_equal
+
+    end = nt._ragged_idx + 1
+    nt_ragged = nt._size[:end]
+    size_ragged = size[:end]
+    if not len(nt_ragged) == len(size_ragged):
+        raise RuntimeError(msg)
+    for ns, s in zip(nt_ragged, size_ragged):
+        if not is_nested_int(s) and s == -1:
+            continue
+        if not (is_nested_int(ns) and is_nested_int(s)):
+            if ns != s:
+                raise RuntimeError(msg)
+            else:
+                continue
+        _nested_assert_metadata_equal(
```

Comment:
I don't actually want this runtime assert unconditionally - maybe need something like in https://github.com/pytorch/pytorch/pull/145957#issuecomment-2632338330, but that is not great.